sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/design/Query_API_Design.md

```diff
@@ -0,0 +1,182 @@
+/// <param name="startTime">Start timestamp of the time range. Events that have this timestamp are included.</param>
+/// <param name="endTime">End timestamp of the time range. Events that match this timestamp are excluded.</param>
+/// <param name="options">Optional parameters to use when querying for aggregate series events.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The pageable list <see cref="AsyncPageable{QueryResultPage}"/> of query result frames.</returns>
+public virtual AsyncPageable<QueryResultPage> QueryAggregateSeriesAsync(
+    TimeSeriesId timeseriesId,
+    DateTimeOffset startTime,
+    DateTimeOffset endTime,
+    QueryAggregateSeriesRequestOptions options = null,
+    CancellationToken cancellationToken = default);
+
+/// <summary>
+/// Retrieve aggregate event series for a given Time Series Id over a certain timespan asynchronously.
+/// </summary>
+/// <param name="timeSeriesId">The Time Series Id to retrieve raw events for.</param>
+/// <param name="timeSpan">The timespan over which to query data.</param>
+/// <param name="options">Optional parameters to use when querying for aggregate series events.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The pageable list <see cref="AsyncPageable{QueryResultPage}"/> of query result frames.</returns>
+public virtual AsyncPageable<QueryResultPage> QueryAggregateSeriesAsync(
+    TimeSeriesId timeseriesId,
+    TimeSpan timeSpan,
+    DateTimeOffset endTime = DateTimeOffset.UtcNow, 
+    QueryAggregateSeriesRequestOptions options = null,
+    CancellationToken cancellationToken = default);
+```
+
+```csharp
+    /// <summary>
+    /// Optional parameters to use when querying for Time Series Insights.
+    /// </summary>
+    public abstract class QueryRequestOptions
+    {
+        /// <summary>
+        /// For the environments with warm store enabled, the query can be executed either on the 'WarmStore' or 'ColdStore'.
+        /// This parameter in the query defines which store the query should be executed on. If not defined, the query
+        /// will be executed on the cold store.
+        /// </summary>
+        public StoreType? StoreType { get; set; }
+
+        /// <summary>
+        /// A Time series expression (TSX) filter written as a single string.
+        /// Refer to the documentation on how to write time series expressions.
+        /// </summary>
+        /// <remarks>
+        /// For filter examples, check out the TSX documentation
+        /// <see href="https://docs.microsoft.com/en-us/rest/api/time-series-insights/reference-time-series-expression-syntax">here.</see>.
+        /// </remarks>
+        public string Filter { get; set; }
+
+        /// <summary>
+        /// The maximum number of property values in the whole response set, not the maximum number of property values per page. Defaults to 10,000 when not set. Maximum value of take can be 250,000.
```

Comment:
I love the comma separator, but remember that in other countries a comma and a period are swapped, so this could look like 250.000. I wonder what the guidance is for documentation WRT this issue with internationalization.

Reply:
Good point. 

I looked at various documentation pages on ms docs. When changing locales, it does differ in terms of whether to display a comma vs a dot. When looking at the source repository for these documentation, they are documented with a comma. I am guessing when generating documentation to be published, they go through some locale appropriate resources to be displayed properly.